neurons/base_node.py

```diff
@@ -61,6 +66,11 @@ async def main(self):
         t.start()
         self._threads.append(t)
 
+        # Start diagnostics task
+        diagnostics_task = asyncio.create_task(self._run_diagnostics())
```

Comment:
have a look at https://github.com/tplr-ai/templar/blob/main/src/tplr/metrics.py#L236

as this will affect the main process loop.